src/NECompletion-Morphic/NECMenuMorph.class.st

```diff
@@ -479,45 +479,10 @@ NECMenuMorph >> initialize [
 	self borderStyle: (BorderStyle color: Color gray width: 1).
 ]
 
-{ #category : #actions }
-NECMenuMorph >> insertCommonPrefixOrSelected [
-	"Return value: true if the user input has been handled, regardless of whether a prefix was inserted"
-
-	| prefix |
-	context hasEntries ifFalse: [^ false].
-	context entries size = 1
-		ifTrue: [ ^ self insertSelected ].
-	prefix := context commonPrefix.
-	prefix = context completionToken ifTrue:[^ self insertSelected].
-	self flag: 'Pending issue 7308, "controller editor wordAtCaret" below should be changed to "context completionToken"'.
-	prefix size > engine editor wordAtCaret size
-		ifTrue: [ 
-			self insertCompletion: prefix.
-			context narrowWith: engine editor wordAtCaret ].
-	^ true
-]
-
-{ #category : #actions }
-NECMenuMorph >> insertCompletion: aString [
-
-	| caret old pos editor offset|
-	editor := engine editor.
-	caret := editor caret.
-	editor selectInvisiblyFrom: caret - context completionToken size to: caret - 1.
-	old := editor selection. 
-	editor replaceSelectionWith: aString.
-	offset := NECPreferences spaceAfterCompletion
-				ifTrue: [ 1 ]
-				ifFalse: [ 0 ].
-	pos := caret + (aString copyUpTo: $ ) size + offset - old size.
-	editor selectAt: pos.
-	editor morph invalidRect: editor morph bounds
-]
-
```

Comment:
Support for insertion in the text editor was moved from the menu morph to the completion engine. See above.